pxd_fastpath.c

```diff
@@ -1404,3 +1455,53 @@ int pxd_switch_nativepath(struct pxd_device* pxd_dev)
 	}
 	return 0;
 }
+
+/// handle io path switch events and io reroute on failures
+/// functions prefixed with ___xxx need to take fail_lock
+static
+void __pxd_add2failQ(struct pxd_device *pxd_dev, struct pxd_io_tracker *iot)
+{
+	list_add_tail(&iot->item, &pxd_dev->fp.failQ);
+}
+
+int __pxd_reissuefailQ(struct pxd_device *pxd_dev, int status)
+{
+	if (!status) {
+		while (!list_empty(&pxd_dev->fp.failQ)) {
+			struct pxd_io_tracker *head = list_first_entry(&pxd_dev->fp.failQ, struct pxd_io_tracker, item);
+			BUG_ON(head->magic != PXD_IOT_MAGIC);
+			list_del(&head->item);
+			if (!pxd_dev->connected || pxd_dev->removing) {
+				printk_ratelimited(KERN_ERR"%s: pxd%llu: px is disconnected, failing IO.\n", __func__, pxd_dev->dev_id);
```

Comment:
This would abort IOs immediately if px is down. IO abort should happen only if px is down for more than 10 mins.

Reply:
yes failure IO path needs rework to hang off abort IO timeout.... will revise.